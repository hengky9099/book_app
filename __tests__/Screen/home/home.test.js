import React from 'react';
import Home from '../../../src/screens/home/index';
import {create} from 'react-test-renderer';
import ContainerTesting from '../../../src/helpers/reduxTesting';

describe('Home Testing', () => {
  describe('Home State Testing', () => {
    const component = create(ContainerTesting(<Home />));
    const root = component.root;
    const {home} = root.props.store.getState();
    test('State Testing', () => {
      expect(root.props.store).toBeTruthy();
      expect(home).toBeTruthy();
      expect(typeof home.recommendeds).toEqual('object');
      expect(typeof home.detailBook).toEqual('object');
    });
  });

  describe('Home Snapshot', () => {
    const tree = create(ContainerTesting(<Home />));
    expect(tree).toMatchSnapshot();
  });
});
