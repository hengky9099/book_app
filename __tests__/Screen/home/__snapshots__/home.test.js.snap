// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    color="#0275d8"
    size="large"
  />
</View>
`;
